apps/console/src/features/roles/components/edit-role/edit-role-updated.tsx

```diff
@@ -0,0 +1,164 @@
+/**
+ * Copyright (c) 2020, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { RoleConstants } from "@wso2is/core/constants";
+import { hasRequiredScopes } from "@wso2is/core/helpers";
+import { RolesInterface, SBACInterface } from "@wso2is/core/models";
+import { ResourceTab, ResourceTabPaneInterface } from "@wso2is/react-components";
+import React, { FunctionComponent, ReactElement, useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useSelector } from "react-redux";
+import { BasicRoleDetails } from "./edit-role-basic-updated";
+import { RoleGroupsList } from "./edit-role-groups-updated";
+import { RolePermissionDetails } from "./edit-role-permission";
+import { RoleUsersList } from "./edit-role-users-updated";
+import { AppState, FeatureConfigInterface } from "../../../core";
+
+/**
+ * Captures props needed for edit role component
+ */
+interface EditRoleProps extends SBACInterface<FeatureConfigInterface> {
+    /**
+     * Is the data loading.
+     */
+    isLoading?: boolean;
+    /**
+     * Role object to be edited.
+     */
+    roleObject: RolesInterface;
+    /**
+     * Callback to update the list of roles.
+     */
+    onRoleUpdate: (activeTabIndex: number) => void;
+    /**
+     * Default active tab index.
+     */
+    defaultActiveIndex?: number;
+}
+
+/**
+ * Component which will allow editing of a selected role.
+ *
+ * @param props - contains role details to be edited.
+ */
+export const EditRole: FunctionComponent<EditRoleProps> = (props: EditRoleProps): ReactElement => {
+
+    const {
+        isLoading,
+        roleObject,
+        onRoleUpdate,
+        defaultActiveIndex
+    } = props;
+
+    const { t } = useTranslation();
+
+    const featureConfig: FeatureConfigInterface = useSelector((state: AppState) => state.config.ui.features);
```

Comment:
Shall we add nullable checks

Reply:
Fixed with https://github.com/wso2/identity-apps/pull/4290/commits/40c5244008b0c41e56b06852cec9b531cdf7b110